src/noflo-component-access.js

```diff
@@ -0,0 +1,138 @@
+ * subset of properties and functions.
+ */
+function facadePort(port) {
+    return _.extend({
+        // port.nodeInstance might not be assigned yet
+        get nodeInstance() {
+            return port.nodeInstance && facadeComponent(port.nodeInstance)
+        },
+        get name() {
+            return port.name;
+        }
+    }, facadeFunctions(port,
+        'getDescription',
+        'getDataType',
+        'hasDefault',
+        'isAddressable',
+        'isBuffered',
+        'isRequired',
+        'isCaching',
+        'isConnected',
+        'connect',
+        'beginGroup',
+        'send',
+        'endGroup',
+        'disconnect',
+        'checkRequired'
+    ), facadeEventEmitter(port));
+}
+
+/**
+ * Returns a facade of the given EventEmitter. The result is not an EventEmitter,
+ * but includes the same functions. Each function call is passed to the underlying
+ * EventEmitter, while any provided functions are bound to the original context
+ * when they are called. This gives the behaviour of an EventEmitter without
+ * actually creating one with a distinct state.
+ */
+function facadeEventEmitter(emitter) {
+    return facadeFunctions(emitter,
+        'addListener',
+        'emit',
+        'getMaxListeners',
+        'listenerCount',
+        'listeners',
+        'on',
+        'once',
+        'removeAllListeners',
+        'removeListener',
+        'setMaxListeners'
+    );
+}
+
+/**
+ * Picks all the given functions from object and returns the set bounded to the
+ * original object context. If any argument is passed to those functions that is
+ * also a function it is bound to the called context.
+ * @param object that contains functions
+ * @param keys explicit names of functions to include in result
+ */
+function facadeFunctions(object /* keys */) {
+    return _.mapObject(_.pick(object, arguments), function(fn) {
+        return _.partial(applyFacade, fn, object)
+    });
+}
+
+/**
+ * If any of the arguments are functions, replace them with a bound version
+ * before calling fn.
+ * @this context that optional function arguments should be bound to
+ * @param fn a function that might take other functions as arguments
+ * @param context that fn should be called with
+ */
+function applyFacade(fn, context /* arguments */) {
+    var args = _.toArray(arguments).slice(2);
+    return fn.apply(context, _.map(args, facadeFunction, this));
+}
+
+/**
+ * Binds the given function to this context or returns the arguments if not a
+ * function. This function will return the same result for repeated calls.
+ * @this context to bind the function to
+ * @param fn a function to bind
+ */
+function facadeFunction(fn) {
+    if (!_.isFunction(fn))
+        return fn;
+    else if (_.isFunction(fn._bind_facade_noflo_access))
+        return fn._bind_facade_noflo_access;
+    else
+        return fn._bind_facade_noflo_access = fn.bind(this);
+}
+
```

Comment:
I kind of like the elegance of your applyFacade and facadeFunction.  Very flexible.